llvm/test/Transforms/EarlyCSE/defaultfp-strictfp.ll

```diff
@@ -246,8 +246,8 @@ define i1 @multiple_fcmp(double %a, double %b) #0 {
 ; CHECK-NEXT:    [[TMP3:%.*]] = call i32 @bar.i32(i32 [[TMP2]], i32 [[TMP2]]) #[[ATTR0]]
 ; CHECK-NEXT:    ret i1 [[TMP1]]
 ;
-  %1 = call i1 @llvm.experimental.constrained.fcmp.i1.f64(double %a, double %b, metadata !"oeq", metadata !"fpexcept.ignore") #0
-  %2 = call i1 @llvm.experimental.constrained.fcmp.i1.f64(double %a, double %b, metadata !"oeq", metadata !"fpexcept.ignore") #0
+  %1 = call i1 @llvm.experimental.constrained.fcmp.f64(double %a, double %b, metadata !"oeq", metadata !"fpexcept.ignore") #0
```

Comment:
Doesn't this still canonically have the i1 output? Or is the vector case sufficiently constrained and understood that the number of elements must match?

Reply:
Yes, the return is specified as `LLVMScalarOrSameVectorWidth<0, llvm_i1_ty>`, so it is not a type parameter. The vector case seems work correctly: https://godbolt.org/z/ozWfvYPMr.